cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -291,16 +279,27 @@ def migrate_from_modulestore(
 
     status.set_state(MigrationStep.POPULATING_COLLECTION.value)
     if target_collection:
-        # @@TODO This dict (block_target_pks_to_change_log_record_pks) was based on the change_log calculation
-        #        above which is commented out. In order to add to target_collection, we either need to fix the
-        #        change_log query above, or we need to calculate these target_pks some other way
-        block_target_pks: list[int] = []  # list(block_target_pks_to_change_log_record_pks.keys())
-        authoring_api.add_to_collection(
-            learning_package_id=target_package_pk,
-            key=target_collection.key,
-            entities_qset=PublishableEntity.objects.filter(id__in=block_target_pks),
-            created_by=user_id,
+        block_target_pks: list[int] = list(
+            ModulestoreBlockMigration.objects.filter(
+                overall_migration=migration
+            ).values_list('target_id', flat=True)
         )
+        if block_target_pks:
+            log.info("Started adding entities to a collection")
+            # @@TODO: Investigate an issue with huge
+            # time consumption during this step
+            # For existing and deleted blocks this step takes
+            # up to 110 seconds
+            # For new blocks this step takes up to 3 seconds
```

Comment:
FYI, I would be OK merging with this part performing poorly as long as we create a follow-up issue to optimize it and link it here.